usb/usb_host_msc/src/msc_host.c

```diff
@@ -296,7 +404,7 @@ esp_err_t msc_host_install(const msc_host_driver_config_t *config)
     MSC_ENTER_CRITICAL();
     MSC_GOTO_ON_FALSE_CRITICAL(!s_msc_driver, ESP_ERR_INVALID_STATE);
     s_msc_driver = driver;
-    STAILQ_INIT(&devices_tailq);
+    STAILQ_INIT(&s_msc_driver->devices_tailq);
```

Comment:
I am still confusing a little bit about this STAILQ.
I was asking myself the same question during the HID implementation but now it is time to ask you this question :)
Why do we need a TAILQ here for devices? 

Probably, to work with any MSC device, connected to the USB port. But MSC device is only the MSC interface within the USB device. 
So, the point is that is we have a USB device, we will work only with first MSC interface, no matter how many MSC interfaces there are present in one device (because the logic of find_msc_interface(..) is to get first one Iface with MSC class and get config from it). 

Probably, the MSC device TAILQ will be reasonable only when:
1. We have an option to launch msc_host_install_device(..) on interface, not on whole USB device based on USB address
2. We pass through all USB device interfaces which can be found in USB descriptor and install all of them (with msc_host_install_device(..))

In the way, how it is implemented right now....seems that this TAILQ will never be containing more that one element.
Only when HUB support will be added,  there is a chance to launch msc_host_install_device(..) twice while one s_msc_driver has been installed. 
But anyway, all the rest MSC interfaces (other than first one) will be skipped.

Probably, that is not a problem for this PR. So it is up to you.

Reply:
> I am still confusing a little bit about this STAILQ.

STAILQ is probably overcomplication here, SLIST would probably suffice here. Comparison of these types can be found in `<sys/queue.h>` header file

> seems that this TAILQ will never be containing more that one element.

This is true for now. When we were designing/ planning this driver, one of the requirements was that it must be future-proof: ready for multiple device support. (There will be proabaly some changes needed, once we get there, though)

> We have an option to launch msc_host_install_device(..) on interface, not on whole USB device based on USB address

I'm not 100% sure about this, but we assumed that if someone needed to implement 'dual MSC' device, the correct way of achieving this would be having one MSC interface with 2 (or more) logical units (LUNs). Which is different to ie CDC, where dual CDC device requires two USB interfaces.

These are all good points. We can follow them up in future

Reply:
Yes, agreed. 
Anyway, both options are possible. 
- LUNs is for distinguishing logical drives within one interface
- Two or more interfaces with MSC device class with 1 LUN

But the second one looks ... strange, that is correct.